### PR TITLE
refactor: Use swagger on Dashboard search API

### PR DESCRIPTION
**Pull request by jinhyukchang**
### Summary of Changes

Migrate Dashboard search API into swagger. 

### Tests

Integration test done.

### Documentation

N/A

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes. 
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does
- [x] PR passes `make test`

### CODE REVIEW

File: tests/unit/api/dashboard/test_search_dashboard_api.py [hunk 1]
```diff
@@ -48,18 +48,6 @@ def test_should_give_empty_result_when_there_are_no_results_from_proxy(self) ->
         }
         self.assertEqual(response.json, expected_response)
 
-    def test_should_get_default_response_values_when_values_not_in_proxy_response(self) -> None:
-        self.mock_proxy.fetch_dashboard_search_results.return_value = \
-            SearchResult(total_results=1, results=[{}])
```
Comment (jinhyukchang) [inline tests/unit/api/dashboard/test_search_dashboard_api.py:53]: Removing this test case for `Dashboard` model fails to instantiate without params.

File: search_service/models/dashboard.py [hunk 2]
```diff
@@ -40,3 +42,15 @@ class DashboardSchema(DashboardSummarySchema):
     class Meta:
         target = Dashboard
         register_as_scheme = True
+
+
+@attr.s(auto_attribs=True, kw_only=True)
```
Comment (ttannis) [inline search_service/models/dashboard.py:47]: What's the reason this can't use the `DashboardSummary`?

Comment (jinhyukchang) [reply search_service/models/dashboard.py:47]: This actually encapsulate `DashboardSummary`.
https://github.com/lyft/amundsensearchlibrary/pull/108/files#diff-7a3f4e6680512c4f1bfa23c826dfd639R13

You can think this is `list` of `DashboardSummary` with `total_result`

Comment (ttannis) [reply search_service/models/dashboard.py:47]: Oh I see it now. Thanks for the explanation.

**Pull request merged**